### DISCUSSION TITLE
My query randomly hangs

### DISCUSSION THREAD

**Discussion by JacobSignal** [supabase/supabase#21895] [Questions]
Hi there i just started using Supabase with Python through SQLAlchemy and psycopg2, I am running a few simple commands to update and insert data into a table hosted on Supabase, where I have a pro account.

However, today I started running into a very strange problem - sometimes my queries randomly hang and then the connection gets closed. Yesterday everything was working great, and if I rerun the query multiple times it has an issue in around 1/10 times otherwise runs perfectly smooth.

The queries are not particularly long, at most its inserting a string of data in the table that has a length of around 3500

I am extremely confused, any tips on how to troubleshoot, what is wrong etc.?

**Comment by GaryAustin1** [comment]
Have you looked thru the dashboard Postgres logs for errors?

**Reply by JacobSignal** [reply]
I am a bit new to Supabase, where can I find that?

**Reply by JacobSignal** [reply]
Just found the logs, I have literally no idea what I should be looking for, there are no errors that really jump out

**Reply by GaryAustin1** [reply]
Right after you get the hang, if there was a timeout or error in the database it should show up as an obvious error in the same time frame in the Postgres logs.   I have no idea though if the database is your issue, that was just a place to start.  I don't use Python so don't have suggestions for debugging the connection side of things.  

Without more info on what is hanging or errors, sort of hard for another user to help you.

**Reply by krugerm** [reply]
I have the same problem - supabase queries randomly hang.  Console shows the network call is never made - I suspect the supabase client is deadlocking when trying take a lock.
This seems to be the same issue as the signOut bug which is also caused by deadlock hang when trying to call signOut().

**Reply by GaryAustin1** [reply]
@krugerm The OP was using Python just to confirm you are using that also, then I have nothing to offer.
If JS then check the important note for the auth event handler https://supabase.com/docs/reference/javascript/auth-onauthstatechange

**Reply by anik8das** [reply]
I'm facing the same issue and this is causing me massive headaches, have you guys found any resolutions? I'm using vite + react
This is my first time using supabase and this is a pathetic experience so far because of this bug

The DB queries randomly hang intermittently
No error log on Supabase, it's as if the request never happened
All of these are simple DB queries, would the OnAuthStateChanged still matter?

**Reply by krugerm** [reply]
OP is using Python, so @anik8das is probably seeing a different issue.
vite / react means you're probably running into this gotcha: https://supabase.com/docs/reference/javascript/auth-onauthstatechange
To resolve, remove all await / promise calls that are inside your onAuthStateChange()

**Reply by anik8das** [reply]
That fixed it, thank you!! @krugerm

**Reply by jdtheefirst** [reply]
@krugerm thanks dude!

**Reply by Hmerac** [reply]
@krugerm Legend. I'm lucky that I found your comment, good luck for the others who are facing the same issue...

**Reply by penguincoder101** [reply]
You are an absolute legend

**Reply by mahdielghali** [reply]
@krugerm Thank you so much!

**Reply by erfannariman** [reply]
@JacobSignal @krugerm did you eventually found anything more for your Python specific question? We think we have the same problem

**Reply by JosephT5566** [reply]
@krugerm thank you so much! I’ve been trying to find a solution to this issue for the past several days.
